### DISCUSSION TITLE
Little Help

### DISCUSSION THREAD

**Discussion by BlutVampir** [discord-php/DiscordPHP#1070] [Q&A] [answered]
I have issues finding the correct PHP Values for some things.

When starting the bot, I use a once Timer to 5s and want to get a list of all threads within a specific channel.

Its a "Board" channel. I need a way to Get a list of all threads which are available in the Board Chanel.
I do have the Board Channel ID.

I did search in all rreps and docs, but cant find a way to get the list of the threads available.

Can anyone help or send me a link to the docs where I can find an idea?

**Comment by BlutVampir** [comment]
$channel = $discord->getChannel('1084862766143246366');

get me:
Discord\Parts\Channel\Channel Object
(
    [id] => 1084862766143246366
    [type] => 15
    [guild_id] => 1084862372948226111
    [position] => 3
    [name] => webchat
    [topic] =>
    [nsfw] =>
    [last_message_id] => 1086273377301450825
    [bitrate] => 64000
    [user_limit] =>
    [rate_limit_per_user] => 0


there you can see last_message_id

but $channel->getMessageHistory(['limit' => 50])->done(function (Collection $messages) {
		print_r($messages);

		foreach ($messages as $message) {
			print_r($message);
		}
	});

gives me nothing. Seams cant get the messages, as they are threads / posts or whatever in the "Board"s

**Comment by BlutVampir** [comment]
Also in Text Channels its not working to get Messages from History:

$channel = $discord->getChannel('1086284827671666828');
$channel->getMessageHistory(['limit' => 50])->done(function (Collection $messages) {
	print_r($messages);

	foreach ($messages as $message) {
		print_r($message);
	}
});

seams bug?

**Reply by CommandString** [reply]
Not sure if the space in your id would cause that issue

**Reply by BlutVampir** [reply]
did correct it, was only a copy tipo. I have an integer in there ;)

**Comment by CommandString** [comment]
[Channel::$threads](https://discord-php.github.io/DiscordPHP/classes/Discord-Parts-Channel-Channel.html#property_threads) not sure if this is automatically filled but I would start by checking that property for getting threads

**Comment by BlutVampir** [comment]
Channel only fills:

ChannelDiscord\Parts\Channel\Channel Object
(
    [id] => 1084862766143246366
    [type] => 15
    [guild_id] => 1084862372948226111
    [position] => 3
    [name] => webchat
    [topic] =>
    [nsfw] =>
    [last_message_id] => 1086281872407019560
    [bitrate] => 64000
    [user_limit] =>
    [rate_limit_per_user] => 0
    [recipients] => Discord\Helpers\Collection Object
        (
        )

    [icon] =>
    [owner_id] =>
    [application_id] =>
    [parent_id] => 1084874292510072902
    [last_pin_timestamp] =>
    [rtc_region] =>
    [video_quality_mode] =>
    [default_auto_archive_duration] =>
    [permissions] =>
    [flags] => 0
    [is_private] =>
)


I can also see it in the code:
 * @property MemberRepository    $members                       Voice channel only - members in the channel.
 * @property MessageRepository   $messages                      Text channel only - messages sent in the channel.
 * @property WebhookRepository   $webhooks                      Webhooks in the channel.
 * @property ThreadRepository    $threads                       Threads that belong to the channel.
 * @property InviteRepository    $invites                       Invites in the channel.

but it is never filled. as all of the "Respository" Thins never get filled.

**Comment by BlutVampir** [comment, marked as answer]
Ohh I found it ;)

$discord->getChannel($webchat)->threads;

not filles, but working like a charm

**Reply by CommandString** [reply]
Was referring to the fact that things like messages aren't not stored in the channel instance automatically (they either have to created when the bot is online or manually fetched from discord) wasn't sure if threads worked this way **but** I'm glad everything is working now!

**Reply by BlutVampir** [reply]
Board posts can be fetches as threads, but you cannot create a new thread, as then the error "this type doesnt support threads" occures. I did chance to a Textchanel to avoid it. But perhaps someone should look into it somewhen, that new "posts" can be made in a board Chanel.
messages and threads are not working there ;)
